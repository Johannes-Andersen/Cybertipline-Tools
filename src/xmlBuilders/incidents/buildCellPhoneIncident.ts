import { XMLBuilder, XMLParser } from 'fast-xml-parser';
import type { CellPhoneIncident } from '../../types';
import { buildPhone } from '../common/buildPhone';

const parser = new XMLParser({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  allowBooleanAttributes: true,
});

const builder = new XMLBuilder({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  suppressBooleanAttributes: false,
});

export const buildCellPhoneIncident = (
  cellPhoneIncident: CellPhoneIncident,
  keyName = 'cellPhoneIncident',
): string =>
  builder.build({
    [keyName]: {
      phoneNumber: cellPhoneIncident.phoneNumber
        ? parser.parse(buildPhone(cellPhoneIncident.phoneNumber)).phone
        : undefined,
      latitude: cellPhoneIncident.latitude,
      longitude: cellPhoneIncident.longitude,
      additionalInfo: cellPhoneIncident.additionalInfo,
    },
  });
