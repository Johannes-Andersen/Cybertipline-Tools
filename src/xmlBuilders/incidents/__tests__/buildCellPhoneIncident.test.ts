import { describe, expect, it } from 'vitest';
import type { CellPhoneIncident } from '../../../types';
import { buildCellPhoneIncident } from '../buildCellPhoneIncident';

describe('buildCellPhoneIncident', () => {
  it('should build a cell phone incident', () => {
    const cellPhoneIncident: CellPhoneIncident = {
      additionalInfo: 'additional info',
      phoneNumber: {
        verified: true,
        number: '1234567890',
      },
      longitude: 456,
      latitude: 123,
    };

    const result = buildCellPhoneIncident(cellPhoneIncident);

    expect(result).toMatchSnapshot();
  });

  it('should build a cell phone incident with no numbers values', () => {
    const cellPhoneIncident: CellPhoneIncident = {
      latitude: 123,
      additionalInfo: 'additional info',
      longitude: 456,
    };

    const result = buildCellPhoneIncident(cellPhoneIncident);

    expect(result).toMatchSnapshot();
  });
});
