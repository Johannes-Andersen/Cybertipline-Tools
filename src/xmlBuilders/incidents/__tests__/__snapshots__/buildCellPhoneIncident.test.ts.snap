// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildCellPhoneIncident > should build a cell phone incident 1`] = `"<cellPhoneIncident><phoneNumber verified="true">1234567890</phoneNumber><latitude>123</latitude><longitude>456</longitude><additionalInfo>additional info</additionalInfo></cellPhoneIncident>"`;

exports[`buildCellPhoneIncident > should build a cell phone incident with no numbers values 1`] = `"<cellPhoneIncident><latitude>123</latitude><longitude>456</longitude><additionalInfo>additional info</additionalInfo></cellPhoneIncident>"`;
